src/managers/notification_manager.ts

```diff
@@ -31,8 +32,12 @@ export default class NotificationManager {
 
 	async handleDataContextChange(/*iNotification: CODAP_Notification*/) {
 		action(async () => {
-			await this.domainStore.featureStore.updateWordListSpecs()
-			await this.domainStore.targetStore.updateFromCODAP()
+			if (!this.updatingStores) {
+				this.updatingStores = true;
+				await this.domainStore.featureStore.updateWordListSpecs()
+				await this.domainStore.targetStore.updateFromCODAP()
+				this.updatingStores = false;
```

Comment:
Is there any chance that either of those function's being awaited will throw an exception which would cause the `this.updatingStores = false` line to not be run?  Maybe it would be best to put this in a try/catch/finally block and set the flag to false in the finally section.

Reply:
Good call 👍